qgis_hub_plugin/gui/resource_browser.py

```diff
@@ -281,6 +293,75 @@ def add_model_to_qgis(self):
                 "model"
             ).refreshAlgorithms()
 
+    def add_geopackage(self):
+        resource = self.selected_resource()
+        file_extension = os.path.splitext(resource.file)[1]
+
+        file_path = QFileDialog.getSaveFileName(
+            self,
+            self.tr("Save Resource"),
+            resource.file,
+            self.tr("ZIP Files (*.zip)"),
+        )[0]
+
+        if file_path:
+            if not file_path.endswith(file_extension):
+                file_path = file_path + file_extension
+
+            if not download_resource_file(resource.file, file_path):
+                self.show_warning_message(
+                    self.tr(f"Download failed for {resource.name}")
+                )
+
+        extract_location = Path(os.path.dirname(file_path))
```

Comment:
The `extract_location` might be not what we expect.
For example:
Resource: QGIS Dashboard
Zip file name: `CasteloBrancoDashboardAnchored_X3UlC1k.zip`
gpkg file name: `CasteloBrancoDashboardAnchored.gpkg`

perhaps try to extract it to a directory, then scan the directory.